### DISCUSSION TITLE
when brew livecheck a cask hosted on github repository, show pre-release in the print mistakenly

### DISCUSSION THREAD

**Discussion by Misaka13514** [Homebrew/discussions#1198] [Everyday usage] [answered]
Until CI, local `brew bump-cask-pr` or `brew audit` didn't report an error.

When I [added livecheck in cask](https://github.com/Homebrew/homebrew-cask/pull/103089), I was told it wasn't necessary, so I thought that something might need to be modified.

**Comment by SMillerDev** [comment, marked as answer]
@samford any suggestions?

**Reply by samford** [reply]
`git ls-remote --tags` (as used in livecheck's `Git` strategy) doesn't tell us whether a tag is marked as pre-release on GitHub, so this isn't really something we can address in livecheck itself. Instead, we handle this for affected formulae/casks using a `livecheck` block.

In this situation, [the newest tag for `classroom-assistant`](https://github.com/education/classroom-assistant/releases) is `2.0.4` but this is "pre-release" on GitHub, so it's necessary to use the `GithubLatest` strategy to check the "latest" release (`2.0.3`). We only use `GithubLatest` when it's both necessary and correct and that's true here.

In my view, Homebrew/homebrew-cask#103089 is the correct approach and should have been merged. I'll comment on that PR to see if they'll re-open and merge it (I don't have commit access to homebrew-cask).

**Reply by samford** [reply]
Homebrew/homebrew-cask#103089 was reopened and merged, so this is taken care of now.
